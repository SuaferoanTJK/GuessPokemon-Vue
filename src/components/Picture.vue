<script>
export default {
  props: {
    pokemonId: { type: Number, required: true },
    showPokemon: { type: Boolean, required: true },
  },
  data() {
    return {
      pokemon: this.pokemonId,
    };
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemon}.svg`;
    },
  },
};
</script>

<template>
  <img
    :src="imgSrc"
    alt="Pokemon"
    class="mx-auto h-64 object-contain transition-all duration-700"
    :class="[showPokemon ? 'brightness-100' : 'brightness-0']"
  />
</template>
