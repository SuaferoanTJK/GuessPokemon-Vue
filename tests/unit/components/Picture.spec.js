import { shallowMount } from "@vue/test-utils";
import Picture from "@/components/Picture";

describe("Picture Unit Testing", () => {
  const props = {
    pokemonId: 1,
    showPokemon: false,
  };
  test("snapshot must be matched", () => {
    const wrapper = shallowMount(Picture, {
      props,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("hide the pokemon if showPokemon=false", () => {
    const wrapper = shallowMount(Picture, {
      props,
    });
    const img = wrapper.find("img");
    expect(wrapper.vm.showPokemon).not.toBeTruthy();
    expect(img.classes("brightness-0")).toBe(true);
    expect(img.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
    );
  });
  test("show the pokemon if showPokemon=true", () => {
    const wrapper = shallowMount(Picture, {
      props: {
        pokemonId: 1,
        showPokemon: true,
      },
    });
    const img = wrapper.find("img");
    expect(wrapper.vm.showPokemon).toBeTruthy();
    expect(img.classes("brightness-100")).toBe(true);
    expect(img.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
    );
  });
});
