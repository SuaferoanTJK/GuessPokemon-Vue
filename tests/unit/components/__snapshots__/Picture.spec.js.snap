// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Unit Testing snapshot must be matched 1`] = `<img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon" class="mx-auto h-64 object-contain transition-all duration-700 brightness-0">`;
